### PR TITLE
protection against edit on  built in strategies

### PR DESCRIPTION
**Pull request by sveisvei**
`unleash-frontend` can then remove the edit mode when editable=false https://github.com/Unleash/unleash-frontend/pull/82

Linked to #209

### CODE REVIEW

File: lib/routes/admin-api/strategy.js [hunk 1]
```diff
@@ -33,6 +33,28 @@ const handleError = (req, res, error) => {
...
+function validateEditable(strategyName) {
+    return strategy => {
+        if (strategy.editable === false) {
+            throw new Error(
+                `Cannot edit strategy ${strategyName}, editable is false`
+            );
+        }
+        return strategy;
+    };
+}
```
Comment (ivarconr) [inline lib/routes/admin-api/strategy.js]: A small unit test of this verification would be awesome!

**Comment by ivarconr** [comment]
Nice, easy and clean! 

I wonder if it would make sense to disable specific "built-in" strategies via config to unleash-server? Should be a different task in that case.

**Comment by sveisvei** [comment]
Should maybe also sort the built-ins last/first.

**Comment by ivarconr** [comment]
We can, but isn't sorting a UI concern?

File: lib/routes/admin-api/strategy.js (hunk 1 above)
Comment (sveisvei) [inline lib/routes/admin-api/strategy.js]: fixed :)

**Pull request merged**